src/exceptiongroup/_exceptions.py

```diff
@@ -5,13 +5,13 @@
 from inspect import getmro, isclass
 from typing import TYPE_CHECKING, Generic, Type, TypeVar, cast, overload
 
-if TYPE_CHECKING:
-    from typing import Self
-
 _BaseExceptionT_co = TypeVar("_BaseExceptionT_co", bound=BaseException, covariant=True)
 _BaseExceptionT = TypeVar("_BaseExceptionT", bound=BaseException)
 _ExceptionT_co = TypeVar("_ExceptionT_co", bound=Exception, covariant=True)
 _ExceptionT = TypeVar("_ExceptionT", bound=Exception)
+# using typing.Self would require a typing_extensions dependency on py<3.11
+_ExceptionGroupSelf = TypeVar("_ExceptionGroupSelf", bound="ExceptionGroup")
+_BaseExceptionGroupSelf = TypeVar("_BaseExceptionGroupSelf", bound="ExceptionGroup")
```

Comment:
Does this really require two different typevars? Why?

Reply:
Oops, set the wrong bound on the second one

Reply:
But yeah we need two different `TypeVar`s, the signature of e.g. `ExceptionGroup.__new__` and `BaseExceptionGroup.__new__` should be different - the former will return a [subclass of] `ExceptionGroup`, the latter a [subclass of] `BaseExceptionGroup`.